lib/express.js

```diff
@@ -33,21 +57,26 @@ exports = module.exports = createApplication;
  * @api public
  */
 
-function createApplication() {
+function createApplication (opts) {
+  var options = opts || {}
+  options.reqProto = options.reqProto || http.IncomingMessage.prototype
+  options.resProto = options.resProto || http.ServerResponse.prototype
+  options.createServer = options.createServer || http.createServer
+
   var app = function(req, res, next) {
     app.handle(req, res, next);
   };
 
   mixin(app, EventEmitter.prototype, false);
   mixin(app, proto, false);
 
+  app.createServer = options.createServer
   // expose the prototype that will get set on requests
-  app.request = Object.create(req, {
+  app.request = Object.create(req(options.reqProto), {
```

Comment:
Why not `Object.create(options.reqProto, ...)` and then `mixin(..., req)` instead? Seems easier to follow and similar performance implications since this is only initialized once.

Reply:
🤷 no idea what I was thinking when I wrote this. I agree that feels more clear to read. I cannot make time to make these changes now but I will when I get time to revisit this pr.